### PR TITLE
dev/core#2364 Use civicrm_api3 and civicrm_api4 instead of civicrm_ap…

### PR DESCRIPTION
**Pull request by seamuslee001**
…i to better handle PEAR_Exceptions

ping @demeritcowboy @eileenmcnaughton

### CODE REVIEW

**Comment by seamuslee001** [comment]
(note that this would need to be ported to Backdrop but can handle once we are comfortable with this)

File: drush/civicrm.drush.inc [hunk 1]
```diff
@@ -1577,9 +1578,31 @@ function drush_civicrm_api() {
...
+      }
+      catch (CiviCRM_API3_Exception $e) {
+        drush_set_error('CIVICRM api error', $e->getMessage());
+      }
+      break;
+
+    case 4:
+      try {
+        unset($params['version']);
+        $restul = civicrm_api4($entity, $action, $params);
```
Comment (demeritcowboy) [inline drush/civicrm.drush.inc]: typo: `restul`

Comment (seamuslee001) [reply drush/civicrm.drush.inc]: dang

File: drush/civicrm.drush.inc [hunk 2]
```diff
@@ -1577,9 +1578,31 @@ function drush_civicrm_api() {
...
+      catch (CiviCRM_API3_Exception $e) {
+        drush_set_error('CIVICRM api error', $e->getMessage());
+      }
+      break;
+
+    case 4:
+      try {
+        unset($params['version']);
+        $result = civicrm_api4($entity, $action, $params);
+      }
```
Comment (demeritcowboy) [inline drush/civicrm.drush.inc:1596]: The PR is all good just noting that depending on the api call the return value from v4 doesn't print anything useful, so e.g. can't really be used for get operations. Long-term is it worth trying to support such a thing from drush?

**Comment by eileenmcnaughton** [comment]
Looks good to me - if @demeritcowboy is happy then good to merge

**Comment by demeritcowboy** [comment]
I'm happy. V4 is no different from before the patch just musing about future.

**Pull request merged**